### PR TITLE
Emit ebpf metrics

### PR DESCRIPTION
**Pull request by mgaurav**
## Summary

Emit ebpf metrics capturing request latency.

### CODE REVIEW

File: datastore/backend.go [hunk 1]
```diff
@@ -52,10 +52,10 @@ func init() {
...
 	}
 
-	MonitoringID = os.Getenv("MONITORING_ID")
-	if MonitoringID == "" {
-		log.Logger.Fatal().Msg("MONITORING_ID is not set")
-	}
+	//MonitoringID = os.Getenv("MONITORING_ID")
+	//if MonitoringID == "" {
+	//	log.Logger.Fatal().Msg("MONITORING_ID is not set")
+	//}
```
Comment (sujeet) [inline datastore/backend.go:61]: would be good to add what this was for and why it makes sense for Oodle to disable this

File: datastore/oodle.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	if isS3PathRegex.MatchString(request.ToUID) {
+		// For S3 requests, strip the bucket name from the path to avoid high cardinality metrics
+		// as path is the object name for S3 calls.
+		// Can use a fingerprint mechanism here to make it generic.
+		parts := strings.SplitN(path, "/", 3)
+		if len(parts) >= 3 {
+			path = "/" + parts[1] // Keep bucket name
+		} else {
+			path = "/" // Default if path doesn't contain enough parts
+		}
```
Comment (sujeet) [inline datastore/oodle.go]: not sure I understood this correctly.
possible to put in a few examples?

for example, I'm not sure what's special about the number 3

File: datastore/oodle.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	}()
+	o.metrics.requestDurationMs.WithLabelValues(
+		path,
+		request.Method,
+		request.Protocol,
+		sourceName,
+		targetName,
+		request.ToType,
+		strconv.Itoa(int(request.StatusCode)),
+		request.FailReason,
```
Comment (sujeet) [inline datastore/oodle.go:159]: general comment, nothing to do with this diff:
this API is not very nice. the declaration is elsewhere and there's no compile-time enforcement of any sort

File: datastore/oodle.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+	if !sOk {
+		sourceName = fmt.Sprintf("%s:%d", request.FromIP, request.FromPort)
+	}
+	targetName, tOk := o.getTargetName(request)
+	if !tOk {
+		targetName = fmt.Sprintf("%s:%d", request.ToIP, request.ToPort)
+	}
+
+	path := request.Path
+	if isS3PathRegex.MatchString(request.ToUID) {
```
Comment (mgaurav) [inline datastore/oodle.go:132]: Add code comment about UID and an example of S3 path name.

**Review by sujeet — APPROVED** [review]

**Pull request merged**